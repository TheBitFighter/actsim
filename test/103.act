deftype str (int<4> x, z)
{
   methods {
     function plus (int<2> y) : int<4>
     {
       chp { self := x + y }
     }
   }
}

defproc test()
{
   str s;
   chp {
      s.x := 3;
      s.z := 1;
      log ("got ", int(s))
   }
}

test t;
