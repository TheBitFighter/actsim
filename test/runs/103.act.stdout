[                  20] <>  got 49
